Lib/test/test_cmd_line.py

```diff
@@ -219,6 +220,23 @@ def check_output(text):
         )
         check_output(text)
 
+    def test_non_interactive_output_buffering(self):
+        def get_value(attr):
+            code = f'import sys; print({attr})'
+            rc, out, err = assert_python_ok('-c', code)
+            return ast.literal_eval(out.rstrip().decode())
+
+        cases = [
+            ('sys.stdout.isatty()', False),
+            ('sys.stdout.write_through', False),
+            ('sys.stdout.line_buffering', False),
+            ('sys.stderr.isatty()', False),
+            ('sys.stderr.write_through', False),
+            ('sys.stderr.line_buffering', True),
+        ]
+        for attr, value in cases:
+            self.assertEqual(get_value(attr), value, f'{attr} is not {value}')
```

Comment:
I suggest to rewrite the test to run Python a single time rather than 6 times, to make the test faster:

```
    def test_non_interactive_output_buffering(self):
        code = textwrap.dedent("""
            import sys
            out = sys.stdout
            print(out.isatty(), out.write_through, out.line_buffering)
            err = sys.stderr
            print(err.isatty(), err.write_through, err.line_buffering)
        """)
        args = [sys.executable, '-c', code]
        proc = subprocess.run(args, stdout=subprocess.PIPE,
                              text=True, check=True)
        self.assertEqual(proc.stdout,
                         'False False False\n'
                         'True False True\n')
```

Reply:
Good point! I used your code, but added `stderr=subprocess.PIPE` to ensure that `sys.stderr` is not a TTY (without this both the old and the new code would pass the test).